benchmark/e2e/test.js

```diff
@@ -0,0 +1,181 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const { spawn, fork } = require('child_process')
+const { promisify } = require('util')
+const { stat } = require('fs')
+const { get: _get } = require('http')
+const path = require('path')
+const mongoService = require('../../packages/dd-trace/test/setup/services/mongo')
+const autocannon = require('autocannon')
+
+function cd (dir) {
+  console.log('> cd', dir)
+  process.chdir(dir)
+}
+
+function delay (ms) {
+  return new Promise((resolve, reject) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+function sh (cmd) {
+  return new Promise((resolve, reject) => {
+    console.log('>', cmd)
+    spawn(cmd, [], { stdio: 'inherit', shell: true })
+      .on('error', reject)
+      .on('close', resolve)
+  })
+}
+
+function forkProcess (file, options = {}) {
+  return new Promise((resolve, reject) => {
+    console.log(`> node ${options.execArgv ? options.execArgv.join(' ') + ' ' : ''}${file}`)
+    options.stdio = 'pipe'
+    const subProcess = fork(file, options)
+    console.log('>> PID', subProcess.pid)
+    subProcess.on('message', message => {
+      if (message.ready) {
+        resolve({ subProcess })
+      }
+    })
+  })
+}
+
+const statAsync = promisify(stat)
+async function exists (filename) {
+  try {
+    const stats = await statAsync(filename)
+    return stats.isDirectory() || stats.isFile()
+  } catch (e) {
+    return false
+  }
+}
+
+function get (url) {
+  return new Promise((resolve, reject) => {
+    _get(url, res => {
+      const chunks = []
+      res.on('data', d => chunks.push(d))
+      res.on('end', () => {
+        resolve(Buffer.concat(chunks).toString())
+      })
+    })
+  })
+}
+
+async function checkDb () {
+  console.log('# checking that db is populated')
+  cd('acmeair-nodejs')
+  const { subProcess } = await forkProcess('./app.js', {
+    execArgv: process.execArgv.concat(['--require', '../datadog.js'])
+  })
+
+  const customers = await get('http://localhost:9080/rest/api/config/countCustomers')
+
+  if (parseInt(customers) < 10000) {
```

Comment:
We'd have to delete the existing collection first, for which there's no handy endpoint. Otherwise it just adds new ones and grows the DB. This is easier.

Reply:
My worry is mostly that it makes this code brittle. What if we change the number? What if for some weird reason the collection is altered? It's safer to always start fresh, especially in tests.

Reply:
Do you want to drop the collection by calling the db directly ahead of time, then?